File: python/insta/guessing_game.py
from random import randrange

d = {"hello": "salama"}

HAND = 10
RANDOM_NUMBER = randrange(1, 100)

print("I am a number between 1 to 99. Try to guess me.")
while HAND > 0:
    guess = int(input("Enter your guess: "))
    if guess == RANDOM_NUMBER:
        print(f"Congratulation. YOU WIN.")
        break
    elif guess > RANDOM_NUMBER:
        print("Try small number.")
    else:
        print("Try larg number.")

    HAND -= 1

if HAND <= 0:
    print("You didn't win this time :-(")


# this is a comment
def hello():
    return "hello"

"""
"""
hello()

File: c/dump/guessing_game.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>



int generateRandNumber(int a) {
	srand(time(0));
	return rand() % a; 
}

int playGame(int randomNumber) {
	int userGuess;
	int guessCount = 0;

	while (userGuess != randomNumber) {
		guessCount ++;
		printf("> ");
		scanf("%d", &userGuess);
		if (userGuess > randomNumber)
			printf("Try smaller!\n");
		else if (userGuess < randomNumber)
			printf("Try bigger!\n");
		else {
			printf("Yay you win with %d tries.\n", guessCount);
			return guessCount;
		}
	} // end of while loop.
} // end of playGame function.

void main() {
	int randomNumber = generateRandNumber(100);
	char ans;

	printf("I am a random number between 0 to 100\n");
	printf("Would you like to guess me (y/n): ");
	scanf("%c", &ans);

	if (ans == 'y') {
		printf("Okey let's play.\n");
		printf("Enter your guess: \n");
		playGame(randomNumber);
	}
	else {
		printf(":-(");
		printf("bye");
	}

} // end of main function.
